tests/storage/restore.go

```diff
@@ -1413,15 +1422,26 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 				vm = createVMWithCloudInit(cd.ContainerDiskCirros, snapshotStorageClass)
 				vm.Spec.Template.Spec.Domain.Firmware = &v1.Firmware{}
 				vm, vmi = createAndStartVM(vm)
+				targetVMName := getTargetVMName(restoreToNewVM, newVmName)
+				login := console.LoginToCirros
 
-				doRestore("", console.LoginToCirros, onlineSnapshot, getTargetVMName(restoreToNewVM, newVmName))
+				if !restoreToNewVM {
+					// Expect to get event in case we stop
+					// the VM after we created the restore.
+					// Once VM is stopped the VMRestore will
+					// continue and complete successfully
+					doRestoreStopVMAfterRestoreCreate("", login, onlineSnapshot, targetVMName)
+				} else {
+					doRestoreNoVMStart("", login, onlineSnapshot, targetVMName)
+				}
+				startVMAfterRestore(targetVMName, "", login)
 				Expect(restore.Status.Restores).To(HaveLen(1))
 				if restoreToNewVM {
 					checkNewVMEquality()
 				}
 
 			},
-				Entry("[test_id:6053] to the same VM", false),
+				Entry("[test_id:6053] to the same VM, stop VM after create restore", false),
```

Comment:
Don't see TargetReadinessPolicy referenced at all? Should we have tests for the different cases

Reply:
I thought UT is enough in this case.. not sure its worth wasting test time for the `fail immediate` case and for the `gracetime` or `eventually` which will require waiting 5 min to check the 1 fails and the other doesnt and allows to restore after the 5 min. If I would add a test might do only for the `stoptarget` one, although if in the UT we see that a Stop call is initiated then Im not sure what a functest will add much to test, but Im not strongly against anything, so if you insist let me know.

Reply:
I think a test with stoptarget would be good

Reply:
Good decision it revealed that an rbac update was required